decoder: Fix timestamps tests

pnx opened the pull request:
<!-- Optional  -->
## Background
Some tests relies on a fixed timezone. making those tests fail on systems that are configured differently.

## Summary
These changes fixes those tests by comparing dates in UTC or setting the timezone temporary in the code. 

## Note
Not sure if this is the correct approach. Maybe the source problem is that the decoder should parse dates as UTC? As far as i know, all time/dates are in UTC in EOS.

## Checklist

- [x] Backward compatible?
- [x] Test enough in your local environment?
- [x] Add related test cases?

maoueh commented on decoder_test.go:
```diff
@@ -439,6 +444,11 @@ func TestDecoder_BlockState(t *testing.T) {
 func TestDecoder_TransactionTrace(t *testing.T) {
+	var err error
+
+	time.Local, err = time.LoadLocation("America/New_York")
+	require.NoError(t, err)
+
```
Can you bring it in https://github.com/eoscanada/eos-go/blob/develop/init_test.go inside the `init` func, this way all tests will have the local timezone changed.

I didn't know it was possible to change it like that, the CI is adjusted to change local timezone in tests, but you are right that it's good to have them work out of the box on developer machine.

pnx replied:
yeah that is a better solution to have it in the `init` func

maoueh commented on decoder_test.go:
```diff
@@ -848,7 +858,7 @@ func TestDecoder_SignedBlock_Full(t *testing.T) {
 	expectedBlockExtension, _ := hex.DecodeString("fe00000000000000000000000000000000000000000000000000000000000004")
 
-	assert.Equal(t, BlockTimestamp{expectedTimestamp}, signedBlock.Timestamp)
+	assert.Equal(t, BlockTimestamp{expectedTimestamp}.UTC(), signedBlock.Timestamp.UTC())
```
Is it still necessary with the modification of the `time.Local` value?

pnx replied:
You are right. with correct timezone converting to UTC is not necessary.

pnx committed:
init_test.go: fixate timezone for tests.

All tests that deals with dates expects the timezone to be in America/New_York.
